adding tmt

Lorquas opened the pull request.

dependabot[bot] committed:
build(deps): bump golang.org/x/sys

Bumps [golang.org/x/sys](https://github.com/golang/sys) from 0.0.0-20210615035016-665e8c7367d1 to 0.1.0.
- [Commits](https://github.com/golang/sys/commits/v0.1.0)

---
updated-dependencies:
- dependency-name: golang.org/x/sys
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

jirihnidek committed:
fix: Update testing matrix

* Drop Go 1.16 from testing matrix
* Added Go 1.20 to the same matrix
* Note: The RHEL7.9 uses Go 1.19 and the latest Fedora uses 1.20

Lorquas committed:
adding tmt

Lorquas commented:
/packit test

Lorquas commented:
/packit-test

Lorquas committed:
removing manual trigger

Lorquas committed:
using internal tf

Lorquas committed:
using internal tf

ptoscano committed:
build: split spec file generation in own target

This way it can be generated separately, without the rest of the srpm
material.

Signed-off-by: Pino Toscano <ptoscano@redhat.com>

ptoscano committed:
build: split tarball generation in own target

This way it can be generated separately, without the rest of the
packaging bits.

Signed-off-by: Pino Toscano <ptoscano@redhat.com>

ptoscano committed:
packit: try to do git builds

ptoscano committed:
simple tmt test for version

ptoscano committed:
build: drop unused replacement for spec file generation

Remove replacements for placeholders that do not exist in rhc.spec.in;
not a functional change, mostly cleanup of unused bits.

Signed-off-by: Pino Toscano <ptoscano@redhat.com>

ptoscano committed:
initial integration test sample

ptoscano committed:
try to pick insights-client

jirihnidek committed:
fix: Fix issues of bash auto-complete script

* Card ID: RHELPLAN-164718
* Checking of issues using shell script analysis tool
  shellcheck should be without any issue now.
* Note: current version of bash auto-complete script provided by
  urfave package looks little bit different, but it is not
  compatible with rhc for some reason.

jstavel committed:
Added an ansible playbook to deploy rhc tests

jstavel committed:
Added an ansible playbook to deploy rhc tests

jstavel committed:
removed ansible stuff - it is not required for simple workflow

jstavel committed:
removed ansible stuff - it is not required for simple workflow

Lorquas commented on systemtest/tests/integration/test.sh:
```diff
@@ -0,0 +1,31 @@
+case "${ID}" in
+  rhel)
+    ;;
+  *)
+    curl --output-dir /etc/yum.repos.d/ -O \
+      "https://ftp.redhat.com/redhat/client-tools/client-tools-for-rhel-${VERSION_MAJOR}.repo"
+    curl -o /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release \
+      https://www.redhat.com/security/data/fd431d51.txt
+    ;;
```
the three cases are 'rhel', 'fedora', 'centos', this logic only works for centos -> should we even try to install insights-client on fedora? If not, the test plugin needs to be able to handle insights-client not existing.

ptoscano replied:
> should we even try to install insights-client on fedora?

yes, however it is not possible at the moment, since insights-client is not available on fedora; this will require making it available on fedora & centos, eg via copr

> the test plugin needs to be able to handle insights-client not existing.

the problem is not much the test plugin, but rather rhc itself that assumes insights-client is always available...

Lorquas commented on systemtest/tests/integration/test.sh:
```diff
@@ -0,0 +1,31 @@
+  subscription-manager \
+  insights-client \
+  podman git-core python3-pip python3-pytest
+
```
would also be nice to be able to run this locally using a tmt container; eg: `tmt run --all provision --how=container --image=quay.io/centos/centos:stream9`

we would need to add the dbus-devel package and this code for this: 
```
if [ ! `command -v rhc` ]; then
  make install
fi
```
That also means we would have to massage the subman commands in the test plugin (since we'll get container mode warnings).  @ptoscano what would your thoughts be on that?  I think it would be nice for developing these tests.

jstavel replied:
I have removed installing python packages directly in test.sh. They will be installed using requirements.txt instead.

ahitacat committed:
fix: insights connection status

This commits fixes the insights-connection status when it is
not connected. Now it shows false when printing in the json
format.

Signed-off-by: Alba Hita Catala <ahitacat@redhat.com>

jstavel committed:
added test for disconnect

jstavel committed:
Merge branch 'packit' of github.com:Lorquas/rhc into packit

Lorquas commented:
Will add candlepin container after: https://github.com/teemtee/tmt/pull/2419

jstavel committed:
feat: all basic tests for rhc

Signed-off-by: Jan Stavel <jstavel@redhat.com>

jstavel committed:
feat: a test for inventory record

Signed-off-by: Jan Stavel <jstavel@redhat.com>

jstavel committed:
feat: a test for tags in inventory

Signed-off-by: Jan Stavel <jstavel@redhat.com>

Lorquas committed:
PEP8

Lorquas committed:
adding pre-commit

Lorquas commented on integration-tests/conftest.py:
```diff
@@ -0,0 +1,80 @@
+        with open("/etc/rhc/tags.toml", "w") as f:
+            toml.dump(tags, f)
+
+    yield _wrapper
+
+    # return the original file back if the one exists
+    if the_config_file_exists:
+        if os.path.isfile(backup_path):
+            shutil.move(backup_path, config_path)
+    else:
+        if os.path.isfile(config_path):
+            os.remove(config_path)
```
because of this yield, the following code does not ever run

jstavel replied:
Hi J.C. are you sure that the code is not called anymore? I guess that is is called in post test stage. since this fixture provides a function instead of static data. This the the only difference with https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#yield-fixtures-recommended

jstavel replied:
fixed problem with global variable the_config_file_exists.

Lorquas commented on integration-tests/test_rhc.py:
```diff
@@ -0,0 +1,136 @@
+
+    with subtests.test(msg="RHC disconnect"):
+        proc = rhc.disconnect()
+        proc_stdout = proc.stdout.decode()
```
you assign `proc_stdout` but never use it?

Lorquas commented on integration-tests/test_rhc.py:
```diff
@@ -0,0 +1,136 @@
+def test_rhc_connect_when_connected(any_candlepin, test_config, rhc, subtests):
+    assert not rhc.is_registered
+    candlepin_config = partial(test_config.get, "candlepin")
+    proc_01 = rhc.connect(
```
you also never use this `proc_01` variable, perhaps:  `_ = rhc.connect( ... `

jstavel committed:
fix: test.sh will use requirements.txt to install the whole python test stack

jstavel committed:
fix: fixtures are returned by yield statement

jstavel committed:
feat: fixture settings for Dynaconf to be available the pytest way

jstavel committed:
Merge branch 'packit' of github.com:Lorquas/rhc into packit

Archana-PandeyM commented on integration-tests/test_rhc.py:
```diff
@@ -11,24 +16,109 @@ def test_version(rhc, subtests):
         logging.info(f'result of disconnect task: {proc.stdout}')
         assert not rhc.is_registered
+        assert "Disconnected from Red Hat Subsription Management",\
+            "The application should inform a system is disconnected from Red Hat Subscription Management"
```
like JC mentioned proc_stdout should be used in assertion, else it wont work as expected.

jstavel replied:
fixed

Archana-PandeyM commented on integration-tests/test_rhc.py:
```diff
@@ -11,24 +16,109 @@ def test_version(rhc, subtests):
+... it will disconnect from all Red Hat services
+... it will inform a user how it goes
+"""
+def test_rhc_connect_disconnect(external_candlepin, test_config, rhc, subtests):
```
Test for activation-key?

Archana-PandeyM committed:
Merge branch 'RedHatInsights:packit' into packit

jstavel committed:
removed mqtt related fixture

jstavel committed:
Merge branch 'packit' of github.com:Lorquas/rhc into packit

jstavel committed:
fix: pre-commit issues fixed

jstavel committed:
feat: rhc connecting using activation key

jstavel committed:
feat: rhc tests use fixture registered_system, not_registered_system already

jstavel committed:
feat: e2e tests uses registered_system and not_registered_system already

jstavel committed:
fix: removed unused dependencies

jstavel committed:
feat: env pytest mark to specify a required dynaconf environment

jstavel committed:
fixed stdout verification

jstavel committed:
added settings fixture, fixed a problem with stdout verification

jstavel committed:
updated to use pytest-client-tools rather than conftest.py
